### ISSUE TITLE
After update: "Server Error or API not found"

### ISSUE THREAD

**Issue by bannisdale** [LycheeOrg/Lychee#884]
### Detailed description of the problem [REQUIRED]
Background: I'm on a managed webserver with Plesk and Softaculous. I updated my instance to the "newest" version available in Softaculous (currently 4.0.8). 

Problem: Now I'm failing to get anywhere – the splash screen shows up but I'm getting the error message "Server Error or API not found" when I click on "gallery"

I tried to enable the debug mode in the config (didn't change anything). Server logs don't seem to show anything suspicious, other than the attempted access in access_ssl_log.

Any hints on how to proceed/analyze further?

### Steps to reproduce the issue
**Steps to reproduce the behavior:**
1. go to URL
2. click on "gallery"
3. empty screen with a red bar with above mentioned message appears

**Screenshot**
![image](https://user-images.githubusercontent.com/42511384/105685409-51aa1980-5ef6-11eb-8ebf-460a8b3d9bd1.png)

### Output of the diagnostics  [REQUIRED]

/Diagnostics shows:

    Diagnostics
    -----------
    Error: Database is behind file versions. Please apply the migration.
    Warning: You may experience problems when uploading a large amount of photos. Take a look in the FAQ for details.
    Warning: Dropbox import not working. dropbox_key is empty.

    System Information
    ------------------
    You must be logged to see this.

    Config Information
    ------------------
    You must be logged to see this.

### Browser and system

Tested in Firefox, Opera and Chrome

**Comment by ildyria** [comment]
Have a look at this: https://lycheeorg.github.io/docs/faq.html#when-i-do-x-i-get-an-error-api-not-found-what-can-i-do

You can also have a look at `storage/log/laravel.log`.

**Comment by bannisdale** [comment]
Thanks for the fast response!

So, I'm getting a Status 500 on Albums::get. FAQ suggests to enable the debug flag in .env. Which didn't help much, so far.

However, the log you suggested might be a little more insightful, even though I can't really make sense of it. :)

```
[previous exception] [object] (PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'smart' in 'where clause' at <host>/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:64)
[stacktrace]
#0 <host>/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php(64): PDO->prepare()
#1 <host>/vendor/laravel/framework/src/Illuminate/Database/Connection.php(331): Doctrine\\DBAL\\Driver\\PDOConnection->prepare()
#2 <host>/vendor/laravel/framework/src/Illuminate/Database/Connection.php(664): Illuminate\\Database\\Connection->Illuminate\\Database\\{closure}()
#3 <host>/vendor/laravel/framework/src/Illuminate/Database/Connection.php(631): Illuminate\\Database\\Connection->runQueryCallback()
#4 <host>/vendor/laravel/framework/src/Illuminate/Database/Connection.php(339): Illuminate\\Database\\Connection->run()
#5 <host>/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2282): Illuminate\\Database\\Connection->select()
#6 <host>/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2270): Illuminate\\Database\\Query\\Builder->runSelect()
#7 <host>/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2765): Illuminate\\Database\\Query\\Builder->Illuminate\\Database\\Query\\{closure}()
#8 <host>/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(2271): Illuminate\\Database\\Query\\Builder->onceWithColumns()
#9 <host>/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(547): Illuminate\\Database\\Query\\Builder->get()
#10 <host>/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(531): Illuminate\\Database\\Eloquent\\Builder->getModels()
#11 <host>/app/ModelFunctions/AlbumFunctions.php(410): Illuminate\\Database\\Eloquent\\Builder->get()
#12 <host>/app/ModelFunctions/AlbumsFunctions.php(213): App\\ModelFunctions\\AlbumFunctions->customSort()
#13 <host>/app/Http/Controllers/AlbumsController.php(64): App\\ModelFunctions\\AlbumsFunctions->getToplevelAlbums()
#14 [internal function]: App\\Http\\Controllers\\AlbumsController->get()
#15 <host>/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): call_user_func_array()
#16 <host>/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction()
#17 <host>/vendor/laravel/framework/src/Illuminate/Routing/Route.php(255): Illuminate\\Routing\\ControllerDispatcher->dispatch()
#18 <host>/vendor/laravel/framework/src/Illuminate/Routing/Route.php(197): Illuminate\\Routing\\Route->runController()
#19 <host>/vendor/laravel/framework/src/Illuminate/Routing/Router.php(691): Illuminate\\Routing\\Route->run()
#20 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
#21 <host>/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(41): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#22 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle()
#23 <host>/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(77): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#24 <host>/app/Http/Middleware/VerifyCsrfToken.php(60): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle()
#25 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): App\\Http\\Middleware\\VerifyCsrfToken->handle()
#26 <host>/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#27 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\AuthenticateSession->handle()
#28 <host>/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#29 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
#30 <host>/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#31 <host>/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(62): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest()
#32 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\StartSession->handle()
#33 <host>/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#34 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle()
#35 <host>/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#36 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
#37 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#38 <host>/vendor/laravel/framework/src/Illuminate/Routing/Router.php(693): Illuminate\\Pipeline\\Pipeline->then()
#39 <host>/vendor/laravel/framework/src/Illuminate/Routing/Router.php(668): Illuminate\\Routing\\Router->runRouteWithinStack()
#40 <host>/vendor/laravel/framework/src/Illuminate/Routing/Router.php(634): Illuminate\\Routing\\Router->runRoute()
#41 <host>/vendor/laravel/framework/src/Illuminate/Routing/Router.php(623): Illuminate\\Routing\\Router->dispatchToRoute()
#42 <host>/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(166): Illuminate\\Routing\\Router->dispatch()
#43 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#44 <host>/vendor/bepsvpt/secure-headers/src/SecureHeadersMiddleware.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#45 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Bepsvpt\\SecureHeaders\\SecureHeadersMiddleware->handle()
#46 <host>/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#47 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#48 <host>/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#49 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#50 <host>/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#51 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#52 <host>/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(87): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#53 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#54 <host>/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#55 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\\Proxy\\TrustProxies->handle()
#56 <host>/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#57 <host>/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(141): Illuminate\\Pipeline\\Pipeline->then()
#58 <host>/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(110): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#59 <host>/public/index.php(74): Illuminate\\Foundation\\Http\\Kernel->handle()
#60 {main}
"} 
```

**Comment by ildyria** [comment]
**Your database is behind your file version.**

Please do `php artisan migrate` and you will be good to go again.
Other possibility is to update the files you have by the one with one of our latest release (4.1+, ideally 4.2+) here: https://github.com/LycheeOrg/Lychee/releases

Upon accessing the website you will be prompted an _update_ page requiring you to enter your admin credentials to apply the migration.

Note that this is the same as this issue: https://github.com/LycheeOrg/Lychee/issues/765
Which is solved coincidentally with version 4.1.0: https://github.com/LycheeOrg/Lychee/pull/798

PS: once fixed, don't forget to change back your `APP_DEBUG` to `false` in your `.env`

**Comment by bannisdale** [comment]
Thanks again for the quick support! :)

I can confirm that the migration was prompted after updating the version to >4.1.0 and that the successful migration made everything running smoothly again.
